src/homeWork2/home_work2.ts

```diff
@@ -62,40 +162,47 @@ class Group {
 class Student {
   // implement 'set grade' and 'set visit' methods
 
-  _firstName;
-  _lastName;
-  _birthYear;
-  _grades = []; // workName: mark
-  _visits = []; // lesson: present
+  private _firstName: string;
+  private _lastName: string;
+  private _birthYear: number;
+  private _grades: (string | number)[] = []; // workName: mark
+  private _visits: boolean[] = []; // lesson: present
 
-  constructor(firstName, lastName, birthYear) {
+  constructor(firstName: string, lastName: string, birthYear: number) {
     this._firstName = firstName;
     this._lastName = lastName;
     this._birthYear = birthYear;
   }
 
-  get fullName() {
+  get fullName(): string {
     return `${this._lastName} ${this._firstName}`;
   }
 
-  set fullName(value) {
-    [this._lastName, this._firstName] = value.split(" ");
+  set fullName(value: string) {
+    [this._lastName, this._firstName] = value.split(' ');
   }
 
-  get age() {
+  get age(): number {
     return new Date().getFullYear() - this._birthYear;
   }
 
-  getPerformanceRating() {
-    const gradeValues = Object.values(this._grades);
+  setGrade(workName: string, mark: number): void {
+    this._grades.push(workName, mark);
+  }
+
+  setVisit(lesson: string, present: boolean): void {
```

Comment:
For what reasons you use lesson param in that fn?

If you want implement such approach you should convert array _visits to an object with string keys "lesson" and boolean value "present".  And than change filter method for _visits by calling "Object.values(_visits)".

Or you can create _visits as array of objects and than change a bit filter method for it.

Or just remove lesson as parameter for that function. You got options:-)